lib/timezone/lookup/geonames.rb

```diff
@@ -36,11 +34,21 @@ def lookup(lat, long)
 
       private
 
+      def get_timezone_id(data)
+        return data['timezoneId'] if data['timezoneId']
+
+        if config.offset_etc_zones && data['gmtOffset']
+          return 'Etc/UTC' if data['gmtOffset'] == 0
+          "Etc/GMT#{format('%+d', data['gmtOffset'].to_i)}"
```

Comment:
I don't think returning an `ArgumentError` is good because the programmer didn't provide a bad argument and this approach would definitely break from the existing API.

At the least we should return `nil` instead of trying to coerce the correct value - as this will have the same effect as having an invalid timezone. If the value is not a Numeric, the integer value will be `0` - which we don't really want.

Maybe something like this?

```
return unless data['gmtOffset'].is_a?(Numeric)
rest of your code as is
```

A test for this scenario would also be good so that the behavior is asserted.